sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/consumer/strategies/synccommit/CommitSyncErrorHandler.java

```diff
@@ -0,0 +1,23 @@
+package org.sdase.commons.server.kafka.consumer.strategies.synccommit;
+
+import org.apache.kafka.clients.consumer.Consumer;
+
+/**
+ * Interface for handling errors that might occur when committing the offset. Be aware that the key
+ * or value can be null if the WrappedNoSerializationErrorDeserializer is used.
+ *
+ * @param <K> key class of the record
+ * @param <V> value class of the record
+ */
+@FunctionalInterface
+public interface CommitSyncErrorHandler<K, V> {
+
+  /**
+   * Handling error that happened during committing.
+   *
+   * @param e error that occurred
+   * @param consumer consumer that read the record
+   * @return if false, the listener not retry to commit the offset.
+   */
+  boolean handleError(RuntimeException e, Consumer<K, V> consumer);
```

Comment:
I get the point that we can't re-use the [ErrorHandler](https://github.com/SDA-SE/sda-dropwizard-commons/blob/297c4b9bfa9bac35aa659d4c512810df1338962c/sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/consumer/ErrorHandler.java) because it's record-based. A specific error handler for sync commits sounds reasonable.